modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -327,39 +384,241 @@ public function insertOrderToDb($id_order, $address, $value, $bits) {
+			return Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		} catch (\Exception $e) {
+				echo "Unable to get order id from blockonomics_bitcoin_orders: {$e->getMessage()}";
+		}
+		return;
+	}
+
+	/*
+	 * Get client currency by invoice id
+	 */	
+	public function getClientCurrency($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		$all_orders_by_id = Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		return $all_orders_by_id;
+	}
+
+	/*
+	 * Check for pending orders and return if exists
+	 */	
+	public function isOrderPending($orders) {
+		foreach ($orders as $order) {
+			//check if status 0 or 1
+			if($order->status == 0 || $order->status == 1){
+				return $order;
+			}
+		}
+		return;
+	}
+
+	/*
+	 * Check for unused address for the currency linked to the order
+	 */		
+	public function isOrderWaiting($orders, $blockonomics_currency) {
+		foreach ($orders as $order) {
+			//check for currency address already waiting
+			if($order->blockonomics_currency == $blockonomics_currency && $order->status == -1){
+				//Check if existing order is expired
+				$current_time = time();
+				$total_time = $this->getTimePeriod() * 60;
+				$clock = $order->timestamp + $total_time - $current_time;
+				if($clock < 0){
+					$order->bits = $this->getNewAmount($order->value, $order->blockonomics_currency);
+					$order->timestamp = $current_time;
+				}
+				$this->updateOrderExpected($order->addr, $order->blockonomics_currency, $order->timestamp, $order->bits);
+				return $order;
+			}
+		}
+		return;
 	}
 
 	/*
-	 * Try to get order row from db by uuid
+	 * Check for unused address or create new
+	 */	
+	public function isOrderNew($orders, $blockonomics_currency) {
+		foreach ($orders as $order) {
+			//check for new blank order in db
+			if($order->addr == ""){
+				$new_addresss_response = $this->getNewAddress($blockonomics_currency);
+				if ($new_addresss_response->response_code == 200){
+					$order->addr = $new_addresss_response->address;
+				}else{
+					exit($new_addresss_response->message);
+				}
+
+				$this->updateOrderAddress($order->id_order, $order->addr, $blockonomics_currency);
+				$order->blockonomics_currency = $blockonomics_currency;
+				$order->bits = $this->getNewAmount($order->value, $order->blockonomics_currency);
+				$order->timestamp = time();
+				$this->updateOrderExpected($order->addr, $order->blockonomics_currency, $order->timestamp, $order->bits);
+				return $order;
+			}
+		}
+		// blank order already used, create a new blank order with same uuid
+		// fix this
+		$order_uuid = $this->newOrder($orders[0]->value, $orders[0]->id_order);
+		$order = $this->getOrderByHash($order_uuid, $blockonomics_currency);
+		return $order;
+	}
+
+	/*
+	 * Get existing or create new order
+	 */	
+	public function getOrderByHash($order_uuid, $blockonomics_currency) {
```

Comment:
- Please rename method name, getOrder implies it is only fetching details. However, this is doing lots of stuff like generating address / setting price
- Rename *order_uuid* variable to order hash as we don't have uuid anymore

Reply:
Updated in 7cf8d9f